src/extension/background-script/connectors/commando.ts

```diff
@@ -0,0 +1,334 @@
+      })
+      .then((resp) => {
+        const parsed = resp as CommandoPayInvoiceResponse;
+        const parsedTotalAmtStr = parsed.amount_msat.replace("msat", "");
+        const parsedSentAmtPaidStr = parsed.amount_sent_msat.replace(
+          "msat",
+          ""
+        );
+        const parsedTotalAmt = (parsedTotalAmtStr as unknown as number) / 1000;
+        const parsedTotalFee =
+          ((parsedSentAmtPaidStr as unknown as number) -
+            (parsedTotalAmtStr as unknown as number)) /
+          1000;
+        return {
+          data: {
+            paymentHash: parsed.payment_hash,
+            preimage: parsed.payment_preimage,
+            route: {
+              total_amt: parsedTotalAmt,
+              total_fees: parsedTotalFee,
+            },
+          },
+        };
+      });
+  }
+
+  async keysend(args: KeysendArgs): Promise<SendPaymentResponse> {
+    //hex encode the record values
+    const records_hex: Record<string, string> = {};
+    for (const key in args.customRecords) {
+      records_hex[key] = UTF8.parse(args.customRecords[key]).toString(Hex);
+    }
+    const boostagram: { [key: string]: string } = {};
+    for (const key in args.customRecords) {
+      boostagram[key] = UTF8.parse(args.customRecords[key]).toString(Hex);
+    }
+    return this.ln
+      .commando({
+        method: "keysend",
+        params: {
+          destination: args.pubkey,
+          msatoshi: args.amount * 1000,
+          extratlvs: boostagram,
+        },
+        rune: this.config.rune,
+      })
+      .then((resp) => {
+        const parsed = resp as CommandoPayInvoiceResponse;
+        const parsedTotalAmtStr = parsed.amount_msat.replace("msat", "");
+        const parsedSentAmtPaidStr = parsed.amount_sent_msat.replace(
+          "msat",
+          ""
+        );
+        const parsedTotalAmt = (parsedTotalAmtStr as unknown as number) / 1000;
+        const parsedTotalFee =
+          ((parsedSentAmtPaidStr as unknown as number) -
+            (parsedTotalAmtStr as unknown as number)) /
+          1000;
+        return {
+          data: {
+            paymentHash: parsed.payment_hash,
+            preimage: parsed.payment_preimage,
+            route: {
+              total_amt: parsedTotalAmt,
+              total_fees: parsedTotalFee,
+            },
+          },
+        };
+      });
+  }
+
+  async checkPayment(args: CheckPaymentArgs): Promise<CheckPaymentResponse> {
+    return this.ln
+      .commando({
+        method: "listinvoices",
+        params: {
+          payment_hash: args.paymentHash,
+        },
+        rune: this.config.rune,
+      })
+      .then((resp) => {
+        const parsed = resp as CommandoListInvoiceResponse;
+        if (parsed.invoices.length != 1) {
+          return {
+            data: {
+              paid: false,
+            },
+          };
+        }
+        return {
+          data: {
+            paid: parsed.invoices[0].status == "paid",
```

Comment:
do we really need that `if (parsed.invoices.length != 1)` check? 

isn't `parsed.invoices[0]?.status == "paid"` enough? 

if there is no invoice it will be undefined and undefined is not equal to paid and such false

Reply:
nit: Try to use `===` or `!==` which also compares the type. Wondering why eslint isn't even showing a warning.